src/main.ts

```diff
@@ -68,6 +79,47 @@ export default class CopyPublishUrlPlugin extends Plugin {
                 }
             },
         });
+        this.registerEvent(this.app.workspace.on("file-menu", (menu, file : TFile) => {

```

Comment:
Please only add this if the user enabled them in the settings. (Off by default.)